### ISSUE TITLE
Upgrade D3

### ISSUE THREAD

**Issue by Xiot** [uber/react-vis#1330]
Upgrade to the latest version of d3

**Comment by chrisirhc** [comment]
I'd think of this as low-pri since d3 has been stable for a while now and upgrading may not bring us any benefit.
Definitely should consider if it gets us any benefits.

**Comment by Xiot** [comment]
Agreed. I just noticed that we've talked behind and might as well update it since we are updating everything else

**Comment by chrisirhc** [comment]
Yep, the other things we're behind actually get us a lot of benefits like yarn workspaces, jest. And there's a lot of development experience improvements.

d3 was already stable when we used it 3-4 years ago. They're essentially utility functions.

Contributions welcome here but this is really low on the list (I'd even consider taking it out of the modernization RFC.

If we double-down on other parts like tests, showcase, documentation we'd get way further.
